test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -940,6 +942,41 @@ public void waitNoPendingTasksOnAll() throws Exception {
         assertNoTimeout(clusterAdmin().prepareHealth(TEST_REQUEST_TIMEOUT).setWaitForEvents(Priority.LANGUID).get());
     }
 
+    /**
+     * Waits for the node {@code viaNode} to see {@code masterNodeName} as the master node in the cluster state and asserts that.
+     * Note that this does not guarantee that all other nodes in the cluster are on the same cluster state version already.
+     *
+     * @param viaNode the node to check the cluster state one
+     * @param masterNodeName the master node name that we wait for
+     */
+    public void awaitAndAssertMasterNode(String viaNode, String masterNodeName) throws Exception {
+        awaitClusterState(
+            logger,
+            viaNode,
+            state -> Optional.ofNullable(state.nodes().getMasterNode()).map(m -> m.getName().equals(masterNodeName)).orElse(false)
+        );
+        var nodes = client(viaNode).admin().cluster().prepareState(TEST_REQUEST_TIMEOUT).setLocal(true).get().getState().nodes();
+        assertThat(Optional.ofNullable(nodes.getMasterNode()).map(DiscoveryNode::getName).orElse(null), equalTo(masterNodeName));
```

Comment:
Unsure why we're using the cluster state API at all here? I think this could technically be flaky, elections don't always settle down on the first attempt so this might not always succeed. Is the `awaitClusterState` not enough?

Reply:
I added the assertion to replace the assertions in `SpecificMasterNodesIT`. Otherwise those tests don't really assert much. They'll indirectly "assert" by waiting for the predicate to match, but I felt like a direct assertion would be more clear. What do you think?

Reply:
I think just waiting for the condition is fine - either the wait completes (which seems sufficient) or it times out (and doesn't even get to this point).